scripts/validate-npm-packages.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script is used to validate the npm packages that are published to npmjs.org are in the correct format.
+# It won't catch things like malformed JS or Types but it will assert that the package has
+# the correct files and package.json properties.
+ARTIFACTS_DIR="./npm-artifacts"
+
+for file in $ARTIFACTS_DIR/*.tgz; do
+  echo "🔍 Checking NPM package: $file"
+
+  dir_name=$(basename "$file" .tgz)
+  mkdir -p "./npm-artifacts/$dir_name"
+  tar -xzf "$file" -C "./npm-artifacts/$dir_name" --strip-components=1
+
+  # Navigate inside the new extracted directory
+  cd "./npm-artifacts/$dir_name"
```

Comment:
consider using `popd` and `pushd` instead of `cd` here and `cd ../..` down the script.